src/main/java/io/nats/client/impl/NatsJetStreamManagement.java

```diff
@@ -62,7 +63,7 @@ private StreamInfo addOrUpdateStream(StreamConfiguration config, String template
 
         String subj = String.format(template, streamName);
         Message resp = makeRequestResponseRequired(subj, config.toJson().getBytes(StandardCharsets.UTF_8), jso.getRequestTimeout());
-        return new StreamInfo(resp).throwOnHasError();
+        return createAndCacheStreamInfo(streamName, resp);
```

Comment:
what happens on an error?

Reply:
renamed the method to indicate it throws on error